File: src/main/java/brs/services/impl/ATServiceImpl.java
package brs.services.impl;

import brs.DependencyProvider;
import brs.at.AT;
import brs.db.store.ATStore;
import brs.services.ATService;

import java.util.Collection;
import java.util.List;

public class ATServiceImpl implements ATService {

  private final DependencyProvider dp;

  public ATServiceImpl(DependencyProvider dp) {
    this.dp = dp;
  }

  @Override
  public Collection<Long> getAllATIds() {
    return dp.atStore.getAllATIds();
  }

  @Override
  public List<Long> getATsIssuedBy(Long accountId) {
    return dp.atStore.getATsIssuedBy(accountId);
  }

  @Override
  public AT getAT(Long id) {
    return dp.atStore.getAT(id);
  }

}

File: src/main/java/brs/services/impl/ATServiceImpl.kt
package brs.services.impl

import brs.DependencyProvider
import brs.at.AT
import brs.services.ATService

class ATServiceImpl(private val dp: DependencyProvider) : ATService {
    override fun getAllATIds() = dp.atStore.getAllATIds()

    override fun getATsIssuedBy(accountId: Long?): List<Long> {
        return dp.atStore.getATsIssuedBy(accountId)
    }

    override fun getAT(id: Long?): AT? {
        return dp.atStore.getAT(id)
    }
}
